app/_how-tos/configure-oidc-with-kong-oauth2.md

```diff
@@ -0,0 +1,180 @@
+We'll use these credentials to generate access tokens.
+
+{% entity_examples %}
+entities:
+  consumers:
+    - username: alex
+      oauth2_credentials:
+        - client_secret: secret
+          client_id: client
+          hash_secret: true
+          name: oauth2-app
+{% endentity_examples %}
+
+## 2. Enable the OAuth2 plugin
+
+The [OAuth2 plugin](/plugins/oauth2/) adds an OAuth 2.0 authentication layer in {{site.base_gateway}} and lets you generate access tokens for Consumers.
+
+First, you'll need a key to provision the plugin. Generate a UUID and export it to an environment variable:
+
+```
+export DECK_PROVISION_KEY=$(uuidgen)
+```
+
+Apply the OAuth2 plugin to the `example-route` Route you created in the [prerequisites](#prerequisites):
+
+{% entity_examples %}
+entities:
+  plugins:
+    - name: oauth2
+      route: example-route
+      config:
+        global_credentials: true
+        enable_client_credentials: true
+        provision_key: ${key}
+variables:
+  key:
+    value: $PROVISION_KEY
+{% endentity_examples %}
+
+## 3. Enable the OpenID Connect plugin with Kong OAuth token authentication
+
+Using the Keycloak and {{site.base_gateway}} configuration from the [prerequisites](#prerequisites), 
+set up an instance of the OpenID Connect plugin with Kong OAuth token authentication.
+
+Enable the OpenID Connect plugin on the `example-service` Service:
+
+{% entity_examples %}
+entities:
+  plugins:
+    - name: openid-connect
+      service: example-service
+      config:
+        issuer: ${issuer}
+        client_id:
+        - ${client-id}
+        client_secret:
+        - ${client-secret}
+        client_auth:
+        - client_secret_post
+        auth_methods:
+        - kong_oauth2
+        bearer_token_param_type:
+        - header
+variables:
+  issuer:
+    value: $ISSUER
+  client-id:
+    value: $CLIENT_ID
+  client-secret:
+    value: $CLIENT_SECRET
+{% endentity_examples %}
+
+In this example:
+* `issuer`, `client ID`, `client secret`, and `client auth`: Settings that connect the plugin to your IdP (in this case, the sample Keycloak app).
+* `auth_methods`: Kong OAuth2 token.
+* `bearer_token_param_type`: We want to search for the token in headers only.
+
+{% include_cached plugins/oidc/client-auth.md %}
+
+## 4. Retrieve the access token
+
+Retrieve the token from the OAuth token endpoint:
+
+```sh
+curl -X POST --insecure https://localhost:8443/anything/oauth2/token \
+  --data "client_id=client" \
+  --data "client_secret=secret" \
+  --data "grant_type=client_credentials"
+```
+
+You should see an `access-token` in the response.
+
+Export the token to an environment variable:
+
+```
+export ACCESS_TOKEN={your-access-token}
+```
+
+## 5. Validate the access token flow
+
+At this point you have created a Gateway Service, routed traffic to the Service, enabled the OpenID Connect plugin, and retrieved the bearer token. 
+Access the `example-route` Route by passing the token you retrieved from the Kong OAuth plugin:
```

Comment:
```suggestion

At this point, you’ve:

* Created a Gateway Service
* Routed traffic to the Service
* Enabled the OpenID Connect plugin
* Retrieved a bearer token

Now, validate the setup by accessing the `example-route` route and including the bearer token you received from the Kong OAuth plugin:

```

You can ignore this one if you don't like it, but it felt nice to break the info up.

Reply:
I think i'd rather just remove the summary. Breaking it up is adding too much extra text to a how-to.